packages/mui-material/src/usePagination/usePagination.test.js

```diff
@@ -10,7 +10,10 @@ describe('usePagination', () => {
   const renderHook = (useHook) => {
     const result = {};
     function TestCase() {
-      result.current = useHook();
+      const hookResult = useHook();
+      React.useEffect(() => {
+        result.current = hookResult;
+      }, [hookResult]);
```

Comment:
Let's undo this change and just silence the ESLint error. This is a test file (the compiler "doesn't matter" here) so let's try to keep it simple and introduce as few changes as possible.